packages/core/src/tw/state/transformation.ts

```diff
@@ -40,6 +40,36 @@ export interface IAppliedTransformation {
     getTransformedRange(): ITransformedRange | undefined;
 }
 
+export class AttributeOperation implements IOperation {
```

Comment:
Following existing naming conventions, I think this should be called `SetAttributesOperation`.

However, there might be better ways to apply attributes. I'm keeping in mind that Taleweaver should support collaborative editing, so ideally, operations should be incremental.

Suppose we want collaborative editing at the attributes level, having an operation that overwrites attributes entirely might be a blocker. Instead, I wonder if we can design the operation to set one attribute at a time? I imagine it's OK to overwrite a single attribute as an edit operation in collaborative editing. For example, if user A sets something to bold and user B sets the same thing to italic, they can both be applied with no conflict. Whereas if user A sets something to bold, and user B removes the bold on the same thing, the collaborative editing algorithm can just pick one as the truth.